context.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2012-2017 Apcera Inc. All rights reserved.
+
+// +build go1.7
+
+// A Go client for the NATS messaging system (https://nats.io).
+package nats
+
+import (
+	"context"
+	"reflect"
+)
+
+// RequestWithContext takes a context, a subject and payload
+// in bytes and request expecting a single response.
+func (nc *Conn) RequestWithContext(ctx context.Context, subj string, data []byte) (*Msg, error) {
+	if ctx == nil {
+		panic("nil context")
```

Comment:
I agree on that, so for now I have removed the checks to not introduce calling `panic` in the nats client either.

Here I was borrowing from the `http`, `net` and other libraries where `panic` is being called...

- https://github.com/golang/go/blob/3792db518327c685da17ca6c6faa4e1d2da4c33c/src/net/dial.go#L316-L319
- https://github.com/golang/go/blob/094498c9a13cd711ed45a65b153393eb8ae1566b/src/net/http/request.go#L323-L334
- https://github.com/go-redis/redis/blob/f33571c93c72d3c4717a7b3c0bce95136496f772/redis_context.go#L28-L32

...though in other packages like  `sql` package those checks aren't introduced either so switched to follow that style instead:

- https://github.com/golang/go/blob/master/src/database/sql/sql.go#L1097-L1118